### ISSUE TITLE
GetApplicationKeyByProcessId throws exception in C#

### ISSUE THREAD

**Issue by BOLL7708** [ValveSoftware/openvr#1306]
I'm running this C# code, using the 1.9.16 header and DLL, with SteamVR beta 1.10.10, on Windows 10 Pro 64-bit (17763):
```Csharp
var pid = OpenVR.Applications.GetCurrentSceneProcessId();
var sb = new StringBuilder();
var error = OpenVR.Applications.GetApplicationKeyByProcessId(pid, sb, OpenVR.k_unMaxApplicationKeyLength);
```
On the third line it crashes out with this:
> An unhandled exception of type 'System.NullReferenceException' occurred in Unknown Module.
> Object reference not set to an instance of an object. occurred

I've tried this in multiple projects which otherwise work just fine, receiving input and reading other data from OpenVR. I'm not sure if I'm missing something here or if it's broken?! 🤷‍♂️

As a side note, if I run this code without any running scene application it works just fine. No exception, no `EVRApplicationError`, and as expected an empty result.
It does crash regardless if I run a a scene application from the Steam client or externally.

**Comment by BOLL7708** [comment]
As a side note to this, something I had happen years ago in yet another application was crashes on `OpenVR.System.GetStringTrackedDeviceProperty()`, which incidentally also takes a `StringBuilder` as the text buffer.  🤔

Interestingly it only crashes for certain properties though, in my old project I found `Prop_SerialNumber_String`, `Prop_HardwareRevision_String` and `Prop_AllWirelessDongleDescriptions_String` to crash for various devices, so I just stopped using them. It does work for other things though, such as `Prop_TrackingSystemName_String`, `Prop_ModelNumber_String` and `Prop_ManufacturerName_String`. Related?
